### PR TITLE
[1857497] [1857498] Validate web app maven plugin configuration with schema

### PR DESCRIPTION
**Pull request by Flanker32**
[AB#1857497](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1857497)
[AB#1857498](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1857498)

*Known Issues*
`[WARNING] Unknown keyword then - you should define your own Meta Schema. If the keyword is irrelevant for validation, just use a NonValidationKeyword`, as external issue of json-schema-validator, refers https://github.com/networknt/json-schema-validator/issues/333

### CODE REVIEW

File: azure-webapp-maven-plugin/src/main/resources/schema/WebAppConfiguration.json [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+    "resourceGroup"
+  ],
+  "definitions": {
+    "non-empty-string": {
+      "type": "string",
+      "minLength": 1
+    },
+    "auth-type": {
+      "type": "string",
+      "pattern": "(?i)^(auto|service_principal|managed_identity|azure_cli|vscode|intellij|azure_auth_maven_plugin|device_code|oauth2|visual_studio)$"
```
Comment (andxu) [inline azure-webapp-maven-plugin/src/main/resources/schema/WebAppConfiguration.json:76]: enum?

File: azure-webapp-maven-plugin/src/main/resources/schema/WebAppConfiguration.json [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+        },
+        "certificatePassword": {
+          "description": "The password for your certificate, if there is any",
+          "type": "string"
+        },
+        "environment": {
+          "description": "The Azure cloud environment",
+          "type": "string",
+          "default": "AZURE",
+          "pattern": "(?i)^(AZURE|AZURE_CHINA|AZURE_GERMANY|AZURE_US_GOVERNMENT)$"
```
Comment (andxu) [inline azure-webapp-maven-plugin/src/main/resources/schema/WebAppConfiguration.json:203]: multiple format of azure environment.

File: azure-webapp-maven-plugin/src/main/resources/schema/WebAppConfiguration.json [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+      "type": "object",
+      "properties": {
+        "type": {
+          "$ref": "#/definitions/auth-type"
+        },
+        "serverId": {
+          "description": "The server id defined in maven settings",
+          "type": "string"
+        },
+        "client": {
```
Comment (andxu) [inline azure-webapp-maven-plugin/src/main/resources/schema/WebAppConfiguration.json:179]: guid

**Review by andxu — APPROVED** [review]

File: azure-webapp-maven-plugin/src/main/resources/schema/WebAppConfiguration.json (hunk 1 above)
Comment (Flanker32) [reply azure-webapp-maven-plugin/src/main/resources/schema/WebAppConfiguration.json:76]: enum in gradle schema is case sensitive, refers https://stackoverflow.com/questions/26604730/can-json-schema-enums-be-case-insensitive

File: azure-webapp-maven-plugin/src/main/resources/schema/WebAppConfiguration.json (hunk 2 above)
Comment (Flanker32) [reply azure-webapp-maven-plugin/src/main/resources/schema/WebAppConfiguration.json:203]: follow document here https://github.com/microsoft/azure-maven-plugins/wiki/Authentication#service-principles-in-plugin-configuration

**Pull request merged**